src/typescript/frontend/src/app/dexscreener/events.ts

```diff
@@ -0,0 +1,314 @@
+  reserves?: {
+    asset0: number | string;
+    asset1: number | string;
+  };
+  metadata?: Record<string, string>;
+}
+
+/**
+ * - `txnId` is a transaction identifier such as a transaction hash
+ * - `txnIndex` refers to the order of a transaction within a block, the higher the index the later in the block the
+ * transaction was processed
+ * - `eventIndex` refer to the order of the event within a transaction, the higher the index the later in the
+ * transaction the event was processed
+ * - The combination of `txnIndex` + `eventIndex` must be unique to any given event within a block, including for
+ * different event types
+ * - `maker` is an identifier for the account responsible for submitting the transaction. In most cases the transaction
+ * is submitted by the same account that sent/received tokens, but in cases where this doesn't apply (i.e.: transaction
+ * submitted by an aggregator) an effort should be made to identify the underlying account.
+ * - All amounts (`assetIn`/`assetOut`/`reserve`) should be decimalized (`amount / (10 ** assetDecimals)`)
+ * - Reserves refer to the pooled amount of each asset *after* a join/exit event has occured. If there are multiple
+ * join/exit events on the same block and reserves cannot be determined after each individual event then it's
+ * acceptable for only the last event to contain the `reserves` prop.
+ * - `metadata` includes any optional auxiliary info not covered in the default schema and not required in most cases
+ */
+export interface JoinExitEvent {
+  eventType: "join" | "exit";
+  txnId: string;
+  txnIndex: number;
+  eventIndex: number;
+  maker: string;
+  pairId: string;
+  amount0: number | string;
+  amount1: number | string;
+  reserves?: {
+    asset0: number | string;
+    asset1: number | string;
+  };
+  metadata?: Record<string, string>;
+}
+
+export type BlockInfo = { block: Block };
+export type Event = (SwapEvent | JoinExitEvent) & BlockInfo;
+
+export interface EventsResponse {
+  events: Event[];
+}
+
+export function toDexscreenerSwapEvent(
+  event: ReturnType<typeof toSwapEventModel>
+): SwapEvent & BlockInfo {
+  let assetInOut;
+
+  if (event.swap.isSell) {
+    // We are selling to APT
+    assetInOut = {
+      asset0In: toCoinDecimalString(event.swap.inputAmount, DECIMALS),
+      asset0Out: 0,
+      asset1In: 0,
+      asset1Out: toCoinDecimalString(event.swap.quoteVolume, DECIMALS),
+    };
+  } else {
+    // We are buying with APT
+    assetInOut = {
+      asset0In: 0,
+      asset0Out: toCoinDecimalString(event.swap.baseVolume, DECIMALS),
+      asset1In: toCoinDecimalString(event.swap.inputAmount, DECIMALS),
+      asset1Out: 0,
+    };
+  }
+
+  const { base, quote } = calculateRealReserves(event.state);
+  const reserves = {
+    asset0: toCoinDecimalString(base, DECIMALS),
+    asset1: toCoinDecimalString(quote, DECIMALS),
+  };
+
+  const priceNative = new Big(event.swap.avgExecutionPriceQ64.toString()).div(2 ** 64).toFixed(64);
```

Comment:
```suggestion
  const priceNative = new Big(event.swap.avgExecutionPriceQ64.toString()).div(2n ** 64n).toFixed(64);
```

I think although you may not run into a loss of precision immediately here, this value is bigger than `Number.MAX_SAFE_INTEGER`, so we should just cast to bigint to be safe